scanspec/service.py

```diff
@@ -62,12 +62,30 @@ class AxisFrames:
     """A fixed reference that can be scanned. i.e. a motor, time or
     number of repetitions.
     """
-    lower: Optional[Points]
+    lower: Points
     """The lower bounds of each midpoint (used when fly scanning)"""
-    midpoints: Optional[Points]
-    """The centre points of the scan"""
-    upper: Optional[Points]
+    midpoints: Points
+    """The middle points of the scan"""
+    upper: Points
     """The upper bounds of each midpoint (used when fly scanning)"""
+    smallest_step: float
+    """The smallest step between midpoints in this axis"""
+
+
+@dataclass
+class SmallestStep:
+    """ The smallest distance between midpoints in a multidimensional scan space"""
+
+    def __init__(self, points: List[ndarray]):
+        # points is an mxn array where n is the number of axes present in the scan
```

Comment:
I guess you mean it is an N-element list of M-element arrays?

Reply:
But isn't `len(points) == m  # number of axes present in the scan (rows)`
and `len(points[i]) == n  # n is the number of points present in each axis`?

So for `Line(y, 0, 1, 10) * Line(x, 0, 1, 5)` then m is 2 and n is 50

Reply:
Oh I understand. Yes, for some reason I had it in my head that it was a list with a single mxn array in it. 

So the list dimension is the columns, (m) being the number of axes present
Then the arrays within the list, (n) are the number of points present in each axis
As you mentioned Tom